tests/fuzz/s2n_recv_client_supported_groups_test.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <stdint.h>
+
+#include <openssl/crypto.h>
+#include <openssl/err.h>
+
+#include "tls/extensions/s2n_client_supported_groups.c"
+
+#include "api/s2n.h"
+#include "stuffer/s2n_stuffer.h"
+#include "tls/s2n_connection.h"
+#include "tls/s2n_tls.h"
+#include "utils/s2n_safety.h"
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+#include "tls/s2n_tls13.h"
+
+/* This test is for TLS versions 1.3 and up only */
+static const uint8_t TLS_VERSIONS[] = {S2N_TLS13};
+
+static void s2n_fuzz_atexit()
+{
+    s2n_cleanup();
+}
+
+int LLVMFuzzerInitialize(const uint8_t *buf, size_t len)
+{
+#ifdef S2N_TEST_IN_FIPS_MODE
+    S2N_TEST_ENTER_FIPS_MODE();
+#endif
+
+    GUARD(s2n_init());
+    GUARD(atexit(s2n_fuzz_atexit));
```

Comment:
`atexit()` returns a non-zero (but not guaranteed to be negative) value on failure. http://man7.org/linux/man-pages/man3/atexit.3.html
`GUARD` checks for negative only.
https://github.com/awslabs/s2n/blob/master/utils/s2n_safety.h#L93

Reply:
Interesting, I was just copying what the other fuzz tests did long before I worked here. Any suggestions for how this should be changed?

Reply:
The simplest would be to change 
```
#define GUARD( x )              do {if ( (x) < 0 ) return S2N_FAILURE;} while (0)
to
#define GUARD( x )              do {if ( (x) != 0 ) return S2N_FAILURE;} while (0)
```

Reply:
Should that be something that I change for just this one file? It seems like that could cause a lot of confusion long-run since we use GUARD throughout the code. Is that something that should be done to change the behavior of GUARD in whole project (in which case we should cut an issue)? If not I'll likely just implement it without the macro for the fuzz tests.

Reply:
That would be in a different PR though, since I'd have to update all the fuzz tests at once.

Reply:
Oh nevermind we already have those guard functions! I'll pop that into these and then make a full change across the fuzz tests in another PR.

Reply:
Created GUARD_STRICT to do just this in another PR and added it to these fuzz tests.